### ISSUE TITLE
MemoryContext error caused by backslash in GUC

### ISSUE THREAD

**Issue by pengzhout** [greenplum-db/gpdb#1301]
```
gpadmin=# set gp_email_to = "\";
ERROR:  problem in alloc set PortalHeapMemory: detected write past chunk end in block 0x2e6a7f0, chunk 0x2e6afb0 (portalmem.c:521) (aset.c:1792)
```

**Comment by hlinnaka** [comment]
PostgreSQL actually happens to have the same bug, although the code has been moved around quite a bit. Reported this at https://www.postgresql.org/message-id/20161216105001.13334.42819%40wrigleys.postgresql.org.

**Comment by hlinnaka** [comment]
Fixed by commit 2dea34a5d39a4ee4bfa07e8195d9e0f931acfcd4.
